src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -73,58 +83,131 @@ public void storeMessage(Message message) {
+    Filter messagesSentByLoggedInUser =
+        new Query.FilterPredicate("user", FilterOperator.EQUAL, loggedInUser);
+    Filter messagesReceivedByOtherUser =
+        new Query.FilterPredicate("recipient", FilterOperator.EQUAL, otherUser);
 
     // All the messages sent by logged-in user to the other user
-    Filter loggedInUserMessages = CompositeFilterOperator.and(messagesSentByLoggedInUser, messagesReceivedByOtherUser);
+    Filter loggedInUserMessages =
+        CompositeFilterOperator.and(messagesSentByLoggedInUser, messagesReceivedByOtherUser);
 
     // Messages between two people, where logged-in user is the recipient
-    Filter messagesSentByOtherUser = new Query.FilterPredicate("user", FilterOperator.EQUAL, otherUser);
-    Filter messagesReceivedByLoggedInUser = new Query.FilterPredicate("recipient", FilterOperator.EQUAL, loggedInUser);
-    
+    Filter messagesSentByOtherUser =
+        new Query.FilterPredicate("user", FilterOperator.EQUAL, otherUser);
+    Filter messagesReceivedByLoggedInUser =
+        new Query.FilterPredicate("recipient", FilterOperator.EQUAL, loggedInUser);
+
     // All the messages recieved by logged-in user sent by the other user
-    Filter otherUserMessages = CompositeFilterOperator.and(messagesSentByOtherUser, messagesReceivedByLoggedInUser);
+    Filter otherUserMessages =
+        CompositeFilterOperator.and(messagesSentByOtherUser, messagesReceivedByLoggedInUser);
 
     // All the messages between the two users
-    Filter directMessages = CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
+    return CompositeFilterOperator.or(loggedInUserMessages, otherUserMessages);
+  }
 
-    // Messages are ordered from oldest to newest since messages between
-    // people may include implicit context/reference to previously sent messages,
-    // and this order makes the chronology of the conversation and context
-    // more intuitive and understandable in the messages list page. 
-    Query query =
-        new Query("Message")
-            .setFilter(directMessages)
-            .addSort("timestamp", SortDirection.ASCENDING);
+  public List<Message> getAllMessages() {
+    Query query = createPublicMessageQuery();
     PreparedQuery results = datastore.prepare(query);
+    return prepareQueryToMesssages(results);
+  }
 
-    for (Entity entity : results.asIterable()) {
-      String idString = entity.getKey().getName();
-      UUID id = UUID.fromString(idString);
-      String text = getStringProperty(entity, "text").orElse("");
-      long timestamp = (long) entity.getProperty("timestamp");
-      String sender = (String) entity.getProperty("user");
-      String receiver = (String) entity.getProperty("recipient");
-      Message message = new Message(id, sender, text, timestamp, receiver);
-      messages.add(message);
+  private List<Message> prepareQueryToMesssages(PreparedQuery query) {
```

Comment:
Minor naming comment: "prepare" -> "prepared". Otherwise, it seems that "prepare" is the verb/action being taken by this function (rather than "preparedQuery" being a single composite noun, referring to the input).